src/cleaner/scripts/clean_docker.py

```diff
@@ -54,44 +55,78 @@ def run(self):
-    def kill_largest_container(self):
+    def kill_largest_container(self, sized, used, usep):
         containers = []
         # Only try to stop PAI jobs and user created containers
-        containers_source = subprocess.Popen(["docker", "ps", "-a", "--format", r'{{.ID}}\t{{.Image}}\t{{.Size}}\t{{.Names}}'], stdout=subprocess.PIPE)
+        containers_source = subprocess.Popen(["docker", "ps", "-a", "--format", r'{{.ID}}\t{{.Image}}\t{{.Size}}\t{{.Names}}\t'], stdout=subprocess.PIPE)
         for line in containers_source.stdout:
             splitline = line.split("\t")
             for prefix in self.white_list:
                 if (splitline[3].startswith(prefix)):
                     break
             else:
-                size = common.calculate_size(splitline[2].split()[0])
-                containers.append([size, splitline[0], splitline[1]])
+                size_str = splitline[2].split()[0]
+                size = common.calculate_size(size_str)
+                containers.append([size, splitline[0], splitline[1], splitline[3], size_str])
 
         containers.sort(key=lambda x:x[0], reverse=True)
 
         if containers.count > 0 and containers[0][0] > 1024**3:
-            self.logger.warning("Kill container {0} due to disk pressure. Container size: {1}".format(containers[0][1], containers[0][0]))
+            self.logger.warning("Kill container {0} due to disk pressure. Container size: {1}".format(containers[0][3], containers[0][4]))
+            
+            # Write error log
+            container_name = re.search(r"container(_\w+)?_\d+_\d+_\d+_\d+$", containers[0][3]).group()
+            application_name = "application{0}".format(re.search(r"^_\d+_\d+", re.search(r"_\d+_\d+_\d+_\d+$", container_name).group()).group())            
+            full_path = "/logs/{0}/{1}".format(application_name, container_name)
+
+            if not os.path.isdir(full_path):
+                self.logger.error("Cannot find job log dir, creating path. Log may not be collected.")
+                try:
+                    os.makedirs(full_path)
+                except OSError as exc:
+                    self.logger.error("Failed to create path {0}.".format(full_path))
+
+            if os.path.isdir(full_path):
+                error_filename = "{0}/diskCleaner.pai.error".format(full_path)
+                timestamp = int(time.time())
+                try:
+                    fp = open(error_filename, "w")
+                except IOError:
+                    self.logger.error("Failed to write error log, skipped")
+                else:
+                    fp.writelines([
+                        "{0} ERROR ACTION \"kill\"\n".format(timestamp),
```

Comment:
Change to "ERROR ACTION \"KILL\"", kill should be converted to uppercase

Reply:
Ok